pipeline.ipynb

```diff
@@ -2,606 +2,618 @@
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 300,
-   "id": "5dba5139-21f0-4e38-9391-c03649a9ec95",
+   "execution_count": 5,
+   "id": "20db0341",
    "metadata": {},
    "outputs": [],
    "source": [
-    "def compute_risk_arrs(start_year, end_year):\n",
-    "    high_risk = []\n",
-    "    low_risk = []\n",
-    "    for year in np.arange(start_year, end_year):\n",
-    "        yr1_fp = met_dir.joinpath(model, scenario, tmp_fn.format(model, scenario, year - 1))\n",
-    "        yr2_fp = met_dir.joinpath(model, scenario, tmp_fn.format(model, scenario, year))\n",
-    "        with xr.open_mfdataset([yr1_fp, yr2_fp]) as ds:\n",
-    "            winter_tmin = ds[\"tmin\"].sel(\n",
-    "                time=slice(f\"{year - 1}-07-01\", f\"{year}-06-30\")\n",
-    "            ).values\n",
-    "            tmax = ds[\"tmax\"].sel(\n",
-    "                time=slice(f\"{year}-01-01\", f\"{year}-12-31\")\n",
-    "            ).values\n",
-    "            tmin = ds[\"tmin\"].sel(\n",
-    "                time=slice(f\"{year}-01-01\", f\"{year}-12-31\")\n",
-    "            ).values\n",
-    "\n",
-    "        # low risk \n",
-    "        # bool for whether or not -40 was exceeded\n",
-    "        cold_arr = winter_tmin.min(axis=0) <= -40\n",
-    "        # bool for whether or not hard cooling happened\n",
-    "        cooling_arr = np.apply_along_axis(hard_cooling, 0, winter_tmin)\n",
-    "        # bool for univoltine summer\n",
-    "        univoltine_arr = np.empty(tmin.shape[1:]).astype(bool)\n",
-    "        for i in np.arange(tmin.shape[1]):\n",
-    "            for j in np.arange(tmax.shape[2]):\n",
-    "                univoltine_arr[i,j] = univoltine(tmin[:,i,j], tmax[:,i,j])\n",
-    "        low_risk.append(cold_arr & cooling_arr & ~univoltine_arr)\n",
-    "\n",
-    "        # high risk\n",
-    "        # warm minimums\n",
-    "        cold_arr = winter_tmin.min(axis=0) > -20\n",
-    "        # will use same univoltine and cooling arrays as above, for now\n",
-    "        high_risk.append(cold_arr & ~cooling_arr & univoltine_arr)\n",
-    "        \n",
-    "    nodata_mask = np.flipud(np.isnan(winter_tmin[0]))\n",
-    "    high_risk = np.flipud(np.array(high_risk).sum(axis=0)).astype(float)\n",
-    "    low_risk = np.flipud(np.array(low_risk).sum(axis=0)).astype(float)\n",
-    "    high_risk[nodata_mask] = np.nan\n",
-    "    low_risk[nodata_mask] = np.nan\n",
-    "    \n",
-    "    return low_risk, high_risk"
+    "job_ids = [slurm.submit_sbatch(fp) for fp in sbatch_fps]"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "6a74a860-b734-4039-ae18-b691c78f9141",
+   "id": "107d0d66",
    "metadata": {},
    "source": [
-    "Compute arrays for the number of \"high risk\" years and \"low risk\" years based on extreme combinations of the three influential criteria."
+    "Note that is seems to be taking **over half an hour** for the processing to complete.\n",
```

Comment:
I ran through this notebook several times and this only took ~8 minutes each time, running on the `main` partition of Atlas.

Reply:
Wow, good to know!